pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -3447,98 +3447,141 @@ private void internalExpireMessagesByTimestampForSinglePartition(String subName,
+                                                    boolean issued;
+                                                    try {
+                                                        if (subName.startsWith(topic.getReplicatorPrefix())) {
+                                                            String remoteCluster = PersistentReplicator
+                                                                    .getRemoteCluster(subName);
+                                                            PersistentReplicator repl = (PersistentReplicator)
+                                                                    topic.getPersistentReplicator(remoteCluster);
+                                                            checkNotNull(repl);
+                                                            issued = repl.expireMessages(position);
+                                                        } else {
+                                                            checkNotNull(sub);
+                                                            issued = sub.expireMessages(position);
+                                                        }
+                                                        if (issued) {
+                                                            log.info("[{}] Message expire started up to {} on {} {}",
+                                                                    clientAppId(), position, topicName, subName);
+                                                        } else {
+                                                            if (log.isDebugEnabled()) {
+                                                                log.debug("Expire message by position not issued on "
+                                                                        + "topic {} for subscription {} due to ongoing "
+                                                                        + "message expiration not finished or "
+                                                                        + "subscription almost catch up.", topicName,
+                                                                        subName);
+                                                            }
+                                                            throw new RestException(Status.CONFLICT, "Expire message by"
+                                                                    + " position not issued on topic " + topicName
+                                                                    + " for subscription " + subName + " due to ongoing"
+                                                                    + " message expiration not finished or invalid "
+                                                                    + "message position provided.");
+                                                        }
+                                                    } catch (NullPointerException npe) {
```

Comment:
Same as https://github.com/apache/pulsar/pull/13876#discussion_r789430042, please help is it necessary for now?